src/adservice/src/main/java/oteldemo/AdService.java

```diff
@@ -160,14 +158,19 @@ public void getAds(AdRequest req, StreamObserver<AdResponse> responseObserver) {
         AdResponseType adResponseType;
 
         Baggage baggage = Baggage.fromContextOrNull(Context.current());
+        MutableContext evaluationContext = new MutableContext();
         if (baggage != null) {
           final String sessionId = baggage.getEntryValue("session.id");
           span.setAttribute("session.id", sessionId);
-          ffClient.setEvaluationContext(new MutableContext().add("session", sessionId));
+          evaluationContext.setTargetingKey(sessionId);
```

Comment:
The targeting key is the only named evaluation context property. When possible, it should be defined because it's used by flagd's deterministic bucketing algorithm by default.